livekit-agents/livekit/agents/voice_assistant/transcription_manager.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import time
+from typing import List
+
+from livekit import rtc
+
+
+class TranscriptionManager:
+    def __init__(self, room: rtc.Room, participant_identity: str):
+        self._participant_identity = participant_identity
+        self._segments: List[rtc.TranscriptionSegment] = []
+        self._track_sid = ""
+        self._room = room
+        self._current_id = 1
+
+    def update_track_sid(self, track_sid: str) -> None:
+        self._track_sid = track_sid
+
+    def reset_transcription(self) -> None:
+        self._segments = []
+
+    def final_transcription(self, text: str, language: str = "english"):
```

Comment:
seems strange to have two diff methods for this? wdyt about `send_transcription(..., final=True)` ?

Reply:
also feels like as an interface.. the following would be cleaner:

```
segment = transcription_manager.start_segment(identity, trackID)
segment.update(full_text, final=True, duration=..., lang=...)
# once final, further .updates would throw an exception
```

It feels like this way, we do not have to create separate instances of TranscriptionManager for every user in the room. feels more natural especially with a dynamic number of speakers